content/chainguard/chainguard-enforce/iam-groups/identity-examples/enforce-buildkite-identity.md

```diff
@@ -0,0 +1,420 @@
+```
+
+The section after that looks up the `viewer` role. 
+
+```
+data "chainguard_roles" "viewer" {
+  name = "viewer"
+}
+```
+
+The final section grants this role to the identity on the `example-group`.
+
+```
+resource "chainguard_rolebinding" "view-stuff" {
+  identity = chainguard_identity.buildkite.id
+  group	= chainguard_group.user-group.id
+  role 	= data.chainguard_roles.viewer.items[0].id
+}
+```
+
+Run the following command to create this file with each of these sections. Be sure to change the `subject_pattern` value to align with your own Buildkite pipeline. For example, if your organization name were `example-org-123` and the pipeline name was `sample-pipeline`, you would set the `subject_pattern` value to `"organization:example-org-123:pipeline:sample-pipeline:ref:refs/heads/main:commit:[0-9a-f]+:step:`.
+
+```sh
+cat > buildkite.tf <<EOF
+resource "chainguard_identity" "buildkite" {
+  parent_id   = chainguard_group.user-group.id
+  name    	= "buildkite"
+  description = <<EOF
+	This is an identity that authorizes Buildkite workflows
+	for this repository to assume to interact with chainctl.
+  EOF
+
+  claim_match {
+	issuer      	= "https://agent.buildkite.com"
+	subject_pattern = "organization:<organization-name>:pipeline:<pipeline-name>:ref:refs/heads/main:commit:[0-9a-f]+:step:"
+  }
+}
+
+output "buildkite-identity" {
+  value = chainguard_identity.buildkite.id
+}
+
+data "chainguard_roles" "viewer" {
+  name = "viewer"
+}
+
+resource "chainguard_rolebinding" "view-stuff" {
+  identity = chainguard_identity.buildkite.id
+  group	= chainguard_group.user-group.id
+  role 	= data.chainguard_roles.viewer.items[0].id
+}
+EOF
+```
+
+Following that, your Terraform configuration will be ready. Now you can run a few `terraform` commands to create the resources defined in your `.tf` files.
+
+
+## Creating Your Resources
+
+First, run `terraform init` to initialize Terraform's working directory.
+
+```sh
+terraform init
+```
+
+Then run `terraform plan`. This will produce a speculative execution plan that outlines what steps Terraform will take to create the resources defined in the files you set up in the last section.
+
+```sh
+terraform plan
+```
+
+If the plan worked successfully and you're satisfied that it will produce the resources you expect, you can apply it.
+
+```sh
+terraform apply
+```
+
+Before going through with applying the Terraform configuration, this command will prompt you to confirm that you want it to do so. Enter `yes` to apply the configuration.
+
+```
+. . .
+
+Plan: 3 to add, 0 to change, 0 to destroy.
+
+Changes to Outputs:
+  + buildkite-identity = (known after apply)
+
+Do you want to perform these actions?
+  Terraform will perform the actions described above.
+  Only 'yes' will be accepted to approve.
+
+  Enter a value:
+```
+
+After pressing `ENTER`, the command will complete and will output an `buildkite-identity` value.
+
+```
+. . .
+
+Apply complete! Resources: 3 added, 0 changed, 0 destroyed.
```

Comment:
Now I run into this - I don't know if it is my Enforce user/group, or something with Terraform, or something else. Did you run into any issues like this one?

```
chainguard_group.user-group: Creating...
chainguard_group.user-group: Creation complete after 0s [id=fcf00e0c8f6a79958403f31a833daee115d63402]
chainguard_policy.cgr-trusted: Creating...
chainguard_identity.buildkite: Creating...
╷
│ Error: rpc error: code = PermissionDenied desc = caller does not have the required capabilities at "fcf00e0c8f6a79958403f31a833daee115d63402", missing: identity.create
│
│   with chainguard_identity.buildkite,
│   on buildkite.tf line 1, in resource "chainguard_identity" "buildkite":
│    1: resource "chainguard_identity" "buildkite" {
│
╵
╷
│ Error: rpc error: code = PermissionDenied desc = caller does not have the required capabilities at "fcf00e0c8f6a79958403f31a833daee115d63402", missing: policy.create
│
│   with chainguard_policy.cgr-trusted,
│   on sample.tf line 10, in resource "chainguard_policy" "cgr-trusted":
│   10: resource "chainguard_policy" "cgr-trusted" {
│
╵
```

Reply:
yes! It's a weird quirk but between `terraform plan` and `terraform apply` you need to `chainctl auth login`. I found a thread about it on slack a while ago (can't find it today) but it's just something you have to do. Something about running `terraform plan` makes you logout of chainctl. I'll add the command here and to the actions guide